### PR TITLE
Android: Migrate to NDK r21.

### LINKED ISSUE #54: Migrate to latest Android LTS ndk (r21)
Should be no real issues here: https://android-developers.googleblog.com/2019/10/introducing-ndk-r21-our-first-long-term.html

### PR DESCRIPTION
**Pull request by Ghabry**
Do not always pull liblcf and Player repo

Fix #54

This strange looking ``./cmdline-tools/latest/bin/sdkmanage`` path is needed. The sdkmanager tells you that it wants this path. sdk_root arg still works but this breaks CMake...

NDK is now fetched via sdk manager :+1: 

Removed:
- standalone toolchain, works now without :)
- That cpufeatures patch was ugly and only fixed 4.0.4 on certain devices so I don't care anymore.

### CODE REVIEW

**Review by fdelapena — APPROVED** [review]

File: android/2_build_toolchain.sh [hunk 1]
```diff
@@ -70,25 +67,24 @@ function build() {
...
+	export CFLAGS="-no-integrated-as -g0 -O2 -fPIC $5"
 	export CXXFLAGS="$CFLAGS -DHB_NO_MMAP"
 	export CPPFLAGS="-I$PLATFORM_PREFIX/include -I$NDK_ROOT/sources/android/cpufeatures"
 	export LDFLAGS="-L$PLATFORM_PREFIX/lib"
 	unset PKG_CONFIG_PATH
 	export PKG_CONFIG_LIBDIR=$PLATFORM_PREFIX/lib/pkgconfig
 	export TARGET_HOST="$4"
-	export CC="$TARGET_HOST-clang"
-	export CXX="$TARGET_HOST-clang++"
+	export CC="clang -target ${4}${TARGET_API}"
```
Comment (carstene1ns) [inline android/2_build_toolchain.sh:80]: `4`→`TARGET_HOST`

File: android/4_build_liblcf.sh [hunk 2]
```diff
@@ -8,11 +8,9 @@ export WORKSPACE=$PWD
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source $SCRIPT_DIR/../shared/import.sh
 
-# Cloning or pulling the liblcf repository
-if [ -d liblcf/.git ]; then
+# Cloning the liblcf repository
+if [ ! -d liblcf/.git ]; then
 	git -C liblcf pull
```
Comment (carstene1ns) [inline android/4_build_liblcf.sh:13]: This makes no sense.
When there is no .git directory there is nothing to pull.

**Review by carstene1ns — CHANGES_REQUESTED** [review]

**Comment by Ghabry** [comment]
Adressed the review comments.
Better don't merge this stuff before Monday. I won't be at home before and Jenkins needs lots of updates for this ^^'

**Review by carstene1ns — APPROVED** [review]

**Comment by Ghabry** [comment]
this is mergeable now

**Pull request merged**